src/PanelTraits/AutoSet.php

```diff
@@ -165,19 +168,37 @@ public function makeLabel($value)
     /**
      * Get the database column names, in order to figure out what fields/columns to show in the auto-fields-and-columns functionality.
      *
-     * @return [array] Database column names as an array.
+     * @param bool $filtered
+     *
+     * @return array Database column names as an array.
      */
-    public function getDbColumnsNames()
+    public function getDbColumnsNames($filtered = true)
     {
-        // Automatically-set columns should be both in the database, and in the $fillable variable on the Eloquent Model
-        $columns = $this->model->getConnection()->getSchemaBuilder()->getColumnListing($this->model->getTable());
-        $fillable = $this->model->getFillable();
+        if (empty($this->db_column_types)) {
+            $this->getDbColumnTypes();
+        }
 
-        if (! empty($fillable)) {
-            $columns = array_intersect($columns, $fillable);
+        //build the db_column_names array only once
+        if (empty($this->db_column_names)) {
+            // Automatically-set columns should be both in the database, and in the $fillable variable on the Eloquent Model
+            $this->db_column_names = array_keys($this->db_column_types);
+
+            if ( $filtered) {
```

Comment:
Ok, maybe that's my problem.

Maybe `$filtered` is just not a good name at all (i.e. I can't figure out what it means - filtered is a tremendously generic word and it doesn't help that datatables and crud panels and such all hijack it to mean something else [i.e. filters on columns]).

Reply:
So, here's the problem... 

The ```setFromDb()``` function is heavily oriented towards editable fields rather than read-only columns for use in 'list' and 'show' views. ```getDbColumnNames()``` strips out all 'columns' that aren't fillable, if the ```fillable``` array is set as well as the timestamp metadata. This is great for autosetting fields for create/update, but not so great for autosetting list/show columns. ```setFromDb()``` only returns fields set by ```getDbColumnNames()``` so fields that aren't 'fillable' are missing. In addition, ```hidden``` fields are allowed in the ```create_fields[]``` and ```edit_fields[]``` arrays but removed from the ```columns[]``` array. This is still the case.

A better name for the ```$filtered = true``` option might be ```$setAllColumns = false```.

An even better option might be to have two new public ```Autoset``` functions:
```setColumnsFromDb()``` and ```setFieldsFromDb()``` which would apply the $filtered/$all option without having to document what it was doing. This would also allow slightly different column-specific logic to be used for determining 'field' type from the db column type. Ideally these would be spawned from the parent ```setFromDb()```. But this would be a potential breaking change since it would globally change the behaviour of ```setFromDb()```

Would that work better?

Reply:
Or we could just document what `$filtered` does in the PHP and on the web-site...

@tabacitu What's your thoughts here?

Reply:
@lloy0076 @tabacitu I think I have a better solution. How about a global config, defaulting to the way things are handled now:
```
 /* file: config/backpack/crud.php */

        // Exclude primary key, 'created', 'updated', 'deleted' metadata for all models in lists by default
        // If this is set to false these columns will be included in list and show views by default.
        //  Note: After setting to false, metadata columns for any individual model can still be excluded by
        //        adding the columns you wish to exclude to the 'hidden' property array of the model
        'exclude_metadata_columns' => true, //current behaviour
```

And then remove '$filtered', leaving the current API and behaviour the same. I have working code for this approach and can submit another pull request.

Reply:
Ugh. Now that you mention it, ```setColumnsFromDb()``` and ```setFieldsFromDb()``` does sound like a cleaner option... and we _could_ call them both from ```setFromDb()``` to make it backwards-compatible...